Marlin/src/module/stepper.cpp

```diff
@@ -2154,6 +2154,7 @@ uint32_t Stepper::block_phase_isr() {
       if ( ENABLED(HAS_L64XX)  // Always set direction for L64xx (Also enables the chips)
         || current_block->direction_bits != last_direction_bits
         || TERN(MIXING_EXTRUDER, false, stepper_extruder != last_moved_extruder)
+        || ENABLED(DUAL_X_CARRIAGE) // For some reason, motion becomes jittery on small circles with dualx machines without this.
```

Comment:
Most likely adding this is slowing down the moves enough to conceal the faster "jitter" moves in-between the direction changes.

Reply:
Is jittery motion also seen when using a single extruder? See if it makes any difference to adjust `MINIMUM_STEPPER_PRE_DIR_DELAY` and `MINIMUM_STEPPER_POST_DIR_DELAY`, since with IDEX duplication and mirror mode we are always setting two DIR, STEP, or EN bit states at the same time.

Reply:
We're using full-control mode, and on prints that use only a single extruder the bug is still seen. Essentially, the dualx feature doesn't even need to be used for the bug to occur.

Reply:
What model of stepper drivers are you using in your testing?

Reply:
TMC2130 standalone

Reply:
Have you compared the behavior of classic jerk to that of linear advance? Are you testing with linear advance on or off?

Reply:
Before this fix we have tried classic jerk vs linear advance to try and remove the layer shift, and since that did not help, we just stuck with linear advance throughout, since it produces better results for our paste.

We have not tried classic jerk with the new layer shift fix.